lib/shelley/bench/restore-bench.hs

```diff
@@ -596,6 +614,52 @@ benchmarksSeq _ w wid _wname benchname restoreTime = do
         , walletOverview
         }
 
+data BenchBaselineResults = BenchBaselineResults
+    { benchName :: Text
+    , restoreTime :: Time
+    } deriving (Show, Generic)
+
+instance Buildable BenchBaselineResults where
+    build = genericF
+
+instance ToJSON BenchBaselineResults where
+    toJSON = genericToJSON Aeson.defaultOptions
+
+{- HLINT ignore bench_baseline_restoration "Use camelCase" -}
+bench_baseline_restoration
+    :: forall (n :: NetworkDiscriminant) .
+        (
+         NetworkDiscriminantVal n
+        , HasNetworkId n
+        )
+    => Proxy n
+    -> Tracer IO (BenchmarkLog n)
+    -> Trace IO Text -- ^ For wallet tracing
+    -> CardanoNodeConn  -- ^ Socket path
+    -> NetworkParameters
+    -> NodeToClientVersionData
+    -> Text -- ^ Benchmark name (used for naming resulting files)
+    -> Bool -- ^ if to write to disk
+    -> Percentage -- ^ Target sync progress
+    -> IO SomeBenchmarkResults
+bench_baseline_restoration proxy tr wlTr socket np vData benchname traceToDisk  targetSync = do
+    putStrLn $ "*** " ++ T.unpack benchname
+    let networkId = networkIdVal proxy
+    let gp = genesisParameters np
+    withWalletLayerTracer benchname traceToDisk $ \progressTrace -> do
+        withNetworkLayer (trMessageText wlTr) networkId np socket vData sTol $ \nw -> do
+            wait <- newEmptyTMVarIO
+            synchronizer <- async $ chainSync nw nullTracer $ ChainFollower
```

Comment:
I recommend to explicitly use record field names when constructing a record value — it will be a bit longer, but the arguments are clearer and easier to disambiguate. This very much applies to `ChainFollower`, whose main purpose is to give names to callback functions. Hence:

```hs
… $ ChainFollower
    { readLocalTip = pure []
    , rollForward = do
        let x = …
    , rollBackward = pure
    } 
```

Reply:
Fixed